src/mapeo-manager.js

```diff
@@ -96,6 +101,20 @@ export class MapeoManager {
       },
     })
 
+    this.#fastifyServer = fastify({ logger: true })
+
+    this.#fastifyServer.decorate(
+      'getProject',
+      /** @param {string} projectId */
+      async (projectId) => {
+        return this.getProject(projectId)
+      }
+    )
+
+    this.#fastifyServer.register(IconServerPlugin, {
+      prefix: '/icons/',
+    })
```

Comment:
didn't realize we were going with a decorator approach, which seems reasonable but maybe not completely necessary imo. with it, looks like we need to declare the dependency on this decorator whenever we register the plugins (https://fastify.dev/docs/latest/Reference/Decorators#decoratename-value-dependencies)

```ts
async function getProjectDecorator (fastify, opts) {
  this.#fastifyServer.decorate(
    'getProject',
    /** @param {string} projectId */
    async (projectId) => {
      return this.getProject(projectId)
    }
  )
}
```

```suggestion
    this.#fastifyServer.register(getProjectDecorator, 'getProject')

    this.#fastifyServer.register(IconServerPlugin, { prefix: '/icons/' }, ['getProject'])
```

Reply:
hm after trying to implement for the blobs plugin, starting to question the benefit of the decorator, especially since you'd have to provide some kind of opt for defining how to get the project anyways. Maybe just keeping it to a `getProject()` opt on the icon/blob plugin is fine as opposed to the decorator?